sklearn/manifold/tests/test_mds.py

```diff
@@ -52,10 +52,98 @@ def test_smacof_error():
     assert_raises(ValueError, mds.smacof, sim, init=Z, n_init=1)
 
 
-def test_MDS():
+def test_fit_error():
+    # fit() should throw an error if the dissimilarity metric is unknown
+    sim = np.array([[0, 5, 3, 4],
+                    [5, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    mds_clf = mds.MDS(dissimilarity="unknown")
+    assert_raises(ValueError, mds_clf.fit, sim)
+
+    mds_clf = mds.MDS(dissimilarity="unknown", extendible=True)
+    assert_raises(ValueError, mds_clf.fit, sim)
+
+
+def test_fit_transform_error():
+    sim = np.array([[0, 5, 3, 4],
+                    [5, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    Z = np.array([[-.266, -.539],
+                  [.451, .252],
+                  [.016, -.238],
+                  [-.200, .524]])
+
+    # fit_transform should throw an error if init is not None
+    # and dissimilarity="euclidean"
+    mds_clf = mds.MDS(dissimilarity="euclidean", extendible=True)
+    assert_raises(ValueError, mds_clf.fit_transform, sim, init=Z)
+
+
+def test_MDS_fit_transform():
     sim = np.array([[0, 5, 3, 4],
                     [5, 0, 2, 2],
                     [3, 2, 0, 1],
                     [4, 2, 1, 0]])
     mds_clf = mds.MDS(metric=False, n_jobs=3, dissimilarity="precomputed")
     mds_clf.fit(sim)
+    mds_clf.fit_transform(sim)
+
+    # Test non-parallel case
+    mds_clf = mds.MDS(metric=False, n_jobs=1, dissimilarity="precomputed")
+    mds_clf.fit(sim)
+    mds_clf.fit_transform(sim)
```

Comment:
Should you be testing that the model is identical?

Reply:
That would be a good improvement. I will add that in.

Reply:
I found there is randomness that cannot be controlled by setting MDS.random_state.  This was true before this PR... I suggest opening an issue and solving that in a follow-up PR?

Reply:
Not yet fixed.

Reply:
Are you accepting that we hold off on fixing this issue?   I looked into it more and the fixes required are out of the scope of this PR.  We can either keep these new tests (checking execution but not result) of different paths using the old/default method of smacof, or I can remove them.

Reply:
Could you please open an issue as I do not feel expert enough in this to understand where the randomness is coming from myself.

Reply:
Sure, no problem.